azerothcore/azerothcore-wotlk
# fix(DB/SAI): Outland Crumbling giants.

Reason for the OOC despawn is that you can potentially lag out an area by spawning ∞ amount of shardlings.

## Changes Proposed:
- Also add OOC (15s) despawn to the Shardlings
- Correct Morcrush hp thresholds & add shardlings spells and despawn OOC (1m)
- Add frenzy and OOC (15s) despawn to Crazed Shardling

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/5022
- Closes https://github.com/chromiecraft/chromiecraft/issues/5021
- Closes https://github.com/chromiecraft/chromiecraft/issues/5016
- Closes https://github.com/chromiecraft/chromiecraft/issues/5006
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14993
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14992
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15000
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14973

## SOURCE:

They don't despawn on summoner's death - https://youtu.be/ltCpiRBlsus?t=131
Mangos/trinity for the serverside spell ids
https://www.youtube.com/watch?v=dHZaZtYUbDA - Morcrush hp thresholds
https://www.wowhead.com/wotlk/npc=21936/crazed-shardling
## Tests Performed:

- tested


## How to Test the Changes:


.go c id 18885
.go c id 18886
.npc add temp 20202 - should also cast periodically Knock Away
.go c id 19940 - should also cast periodically Thunderclap
.go c id 20772
.go c id 18690 - its shardling should cast thunderclap and enrage at 50%
.npc add temp 21936 should cast frenzy

See if they become smaller, deal less damage and spawn 1x Shardling on the hp thresh holds 75/50/25, for 20202 is 90/60/30.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
